### PR TITLE
implement session token support for Azure federated credentials

### PR DESCRIPTION
**Pull request by shenmo3**
## Description
This PR implements support for session token in Azure CloudProviderAccount config. The session token can be used for Azure federated credentials in which the clientKey field should be omitted.

## Changes
1. Add sessionToken field in AzureAccountCredential for k8s secrets unmarshal.
2. Add Azure client credential handling for session token based auth.
3. Add unit test for session token based CPA.
4. Update doc to include session token.

### CODE REVIEW

File: pkg/apiserver/webhook/cloudprovideraccount_webhook.go [hunk 1]
```diff
@@ -40,17 +40,18 @@ import (
...
-	errorMsgInvalidRegion        = "not in supported regions"
-	errorMsgJsonUnmarshalFail    = "unable to unmarshal the json"
-	errorMsgMissingClientDetails = "client id and client key cannot be blank or empty"
-	errorMsgMissingTenantID      = "tenant id cannot be blank or empty"
-	errorMsgMissingSubscritionID = "subscription id cannot be blank or empty"
-	errorMsgInvalidRequest       = "invalid admission webhook request"
-	errorMsgDecodeFail           = "unable to decode the secret"
+	errorMsgSecretNotConfigured    = "unable to get secret"
+	errorMsgMinPollInterval        = "pollIntervalInSeconds should be >= 30. If not specified, defaults to 60"
+	errorMsgMissingAwsCredential   = "must specify either credentials or role arn, cannot both be empty"
```
Comment (reachjainrahul) [inline pkg/apiserver/webhook/cloudprovideraccount_webhook.go:45]:
```suggestion
	errorMsgMissingAwsCredential   = "must specify either credentials or role arn, both cannot be empty"
```

File: pkg/apiserver/webhook/cloudprovideraccount_webhook.go [hunk 2]
```diff
@@ -40,17 +40,18 @@ import (
...
+	errorMsgMissingRegion          = "region cannot be blank or empty"
+	errorMsgInvalidRegion          = "not in supported regions"
+	errorMsgJsonUnmarshalFail      = "unable to unmarshal the json"
+	errorMsgMissingClientID        = "client id cannot be blank or empty"
+	errorMsgMissingAzureCredential = "must specify either client key or session token, cannot both be empty"
```
Comment (reachjainrahul) [inline pkg/apiserver/webhook/cloudprovideraccount_webhook.go:50]: same here

File: pkg/cloudprovider/plugins/azure/azure_test.go [hunk 3]
```diff
@@ -177,8 +212,19 @@ var _ = Describe("Azure", func() {
 						"credentials": []byte(credential),
 					},
 				}
+				_ = fakeClient.Update(context.Background(), secret)
+
+				c := newAzureCloud(mockAzureServiceHelper)
 
-				_ = fakeClient.Create(context.Background(), secret)
+				err := c.AddProviderAccount(fakeClient, account)
```
Comment (reachjainrahul) [inline pkg/cloudprovider/plugins/azure/azure_test.go:219]: could you make sure this is not retry-able error

File: pkg/apiserver/webhook/cloudprovideraccount_webhook.go (hunk 1 above)
Comment (shenmo3) [reply pkg/apiserver/webhook/cloudprovideraccount_webhook.go]: Done

File: pkg/apiserver/webhook/cloudprovideraccount_webhook.go (hunk 2 above)
Comment (shenmo3) [reply pkg/apiserver/webhook/cloudprovideraccount_webhook.go]: Done

File: pkg/cloudprovider/plugins/azure/azure_test.go (hunk 3 above)
Comment (shenmo3) [reply pkg/cloudprovider/plugins/azure/azure_test.go:220]: Done

**Review by reachjainrahul — APPROVED** [review]
/LGTM

**Pull request merged**